### ISSUE TITLE
[BUG]: Timeseries forecasting ignores passed timerange and uses test set timerange (cutoff date)

### ISSUE THREAD

**Issue by Baukebrenninkmeijer** [pycaret/pycaret#4112]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

When using `predict_model` on a dataframe with timeseries data with exogenous variables, it seems the forecasting ignores the original time index. What i'm expecting to happen is that I get a forecast/prediction for each timestep that is in the index. Perhaps i'm misunderstanding the intended usage, but that's my current understanding of what should happen. 

### Reproducible Example

```python
from pycaret.datasets import get_data
from pycaret.time_series import TSForecastingExperiment

airline_df = get_data('airline')

mini_exp = TSForecastingExperiment()
mini_exp.setup(data=airline_df, session_id=0, fh=1)
top3 = mini_exp.compare_models(exclude=['auto_arima'], n_select=3)
leaderboard_mini = mini_exp.get_leaderboard()


best_model_unfinalized = leaderboard_mini.sort_values('SMAPE')['Model'].iloc[0].steps[1][1] # apparently, this gets me the actual model. Otherwise it returns a dummy forecaster, or some other non-functional predictor. Any insights as to why this happens are also greatly appreciated. 

best_model_finalized = mini_exp.finalize_model(leaderboard_mini.sort_values('SMAPE')['Model'].iloc[0].steps[1][1])


lynx_df = get_data('lynx') # Get inference dataset with unseen data. In this case, completely unrelated data. 

mini_exp.plot_model([best_model_finalized, best_model_unfinalized], data_kwargs={'fh': 10, 'X': lynx_df}) # the X data passed here is a completely different dataset
```

### Expected Behavior

This throws an error. The dataset has a different frequency and different exogenous variables. This should not succeed.

### Actual Results

```python-traceback
Code runs fine, but ignores time index of the inference data. My target use-case is to have pycaret predict an arbitrary timerange that is not immediately before or after the cutoff date. I thought passing X for this range might work, but this is essentially ignored? I'm quite unsure what is actually happening and how to get the behaviour that I desire.
```

### Installed Versions

<details>

System:
    python: 3.11.9 (main, Aug 28 2024, 08:57:33) [Clang 15.0.0 (clang-1500.3.9.4)]
executable: /Users/tn14gn/.pyenv/versions/pss-311/bin/python
   machine: macOS-15.1.1-arm64-arm-64bit

PyCaret required dependencies:
                 pip: 24.0
          setuptools: 75.6.0
             pycaret: 3.3.2
             IPython: 8.30.0
          ipywidgets: 8.1.2
                tqdm: 4.67.1
               numpy: 1.26.4
              pandas: 2.1.4
              jinja2: 3.1.4
               scipy: 1.11.4
              joblib: 1.3.2
             sklearn: 1.4.2
                pyod: 2.0.2
            imblearn: 0.12.0
   category_encoders: 2.6.4
            lightgbm: 4.5.0
               numba: 0.60.0
            requests: 2.32.3
          matplotlib: 3.7.5
          scikitplot: 0.3.7
         yellowbrick: 1.5
              plotly: 5.19.0
    plotly-resampler: Not installed
             kaleido: 0.2.1
           schemdraw: 0.15
         statsmodels: 0.14.2
              sktime: 0.26.0
               tbats: 1.1.3
            pmdarima: 2.0.4
              psutil: 6.1.0
          markupsafe: 2.1.5
             pickle5: Not installed
         cloudpickle: 3.0.0
         deprecation: 2.1.0
              xxhash: 3.5.0
           wurlitzer: 3.1.1

PyCaret optional dependencies:
                shap: Not installed
           interpret: Not installed
                umap: 0.5.7
     ydata_profiling: Not installed
  explainerdashboard: Not installed
             autoviz: Not installed
           fairlearn: Not installed
          deepchecks: Not installed
             xgboost: 2.0.3
            catboost: Not installed
              kmodes: Not installed
             mlxtend: Not installed
       statsforecast: Not installed
        tune_sklearn: Not installed
                 ray: Not installed
            hyperopt: Not installed
              optuna: Not installed
               skopt: Not installed
              mlflow: Not installed
              gradio: Not installed
             fastapi: Not installed
             uvicorn: Not installed
              m2cgen: Not installed
           evidently: Not installed
               fugue: Not installed
           streamlit: 1.41.1
             prophet: 1.1.5
</details>

**Comment by Baukebrenninkmeijer** [comment]
I rather not install the main branch btw, I just got all dependencies figured out and everything running, and that was not an easy task. Doubt this issue was recently fixed, as I am on the latest version. Let me know if it's required please.

**Comment by chengzhaoan** [comment]
me too

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
